scrapy/crawler.py

```diff
@@ -72,42 +72,55 @@ def __init__(
         self.spidercls: Type[Spider] = spidercls
         self.settings: Settings = settings.copy()
         self.spidercls.update_settings(self.settings)
+        self._update_root_log_handler()
 
         self.addons: AddonManager = AddonManager(self)
-        self.addons.load_settings(self.settings)
-
         self.signals: SignalManager = SignalManager(self)
 
-        self.stats: StatsCollector = load_object(self.settings["STATS_CLASS"])(self)
-
-        handler = LogCounterHandler(self, level=self.settings.get("LOG_LEVEL"))
-        logging.root.addHandler(handler)
+        self._init_reactor: bool = init_reactor
+        self.crawling: bool = False
+        self._settings_loaded: bool = False
+        self._started: bool = False
 
-        d = dict(overridden_settings(self.settings))
-        logger.info(
-            "Overridden settings:\n%(settings)s", {"settings": pprint.pformat(d)}
-        )
+        self.extensions: Optional[ExtensionManager] = None
+        self.stats: Optional[StatsCollector] = None
+        self.logformatter: Optional[LogFormatter] = None
+        self.request_fingerprinter: Optional[RequestFingerprinter] = None
+        self.spider: Optional[Spider] = None
+        self.engine: Optional[ExecutionEngine] = None
 
+    def _update_root_log_handler(self) -> None:
         if get_scrapy_root_handler() is not None:
             # scrapy root handler already installed: update it with new settings
             install_scrapy_root_handler(self.settings)
+
+    def _load_settings(self) -> None:
```

Comment:
a nitpick, about the method name :) _load_settings suggests that this method loads the settings (from somewhere?). It seems what this method does is to load/create the components based on the settings, not loading the settings themselves.

This seems to apply to AddonManager.load_settings name as well, btw. Not sure if it's too late to change this; it seems we could, as we haven't made AddonManager a part of the public API.

Some name suggestions:

* _apply_settings

or, if -apply_settings doesn't sound right, something like 

* _load_components
* _create_components

Reply:
_apply_settings is fine and was proposed before